libraries/AP_Arming/AP_Arming.cpp

```diff
@@ -887,6 +892,20 @@ bool AP_Arming::servo_checks(bool report) const
                 check_passed = false;
             }
         }
+
+#if HAL_WITH_IO_MCU && (APM_BUILD_COPTER_OR_HELI || APM_BUILD_TYPE(APM_BUILD_ArduPlane) || APM_BUILD_TYPE(APM_BUILD_Rover))
+        if (i < 8 && ch_function >= SRV_Channel::k_motor1 && ch_function <= SRV_Channel::k_motor8) {
+#if APM_BUILD_TYPE(APM_BUILD_Rover)
+            AP_MotorsUGV *motors = AP::motors_ugv();
+#else
+            AP_Motors *motors = AP::motors();
+#endif
+            if (motors && motors->is_digital_pwm_type()) {
+                check_failed(report, "SERVO%u_FUNCTION set to Motor%u on non-digital channel", unsigned(ch_function), i + 1);
```

Comment:
This generally looks OK but for Rover, the MotorX servo functions are only used for omni vehicles.  Is it really only these servo functions that can be used with dshot?  I would have thought that Throttle, ThrottleLeft, ThrottleRight could also be enabled to use dshot.

Reply:
Ok, I have done this a different way